embedder/cohere/cohere.go

```diff
@@ -0,0 +1,65 @@
+package cohereembedder
+
+import (
+	"context"
+	"github.com/cohere-ai/cohere-go"
+	"github.com/henomis/lingoose/embedder"
+	"net/http"
+)
+
+type Model string
+
+const EmbedMultilingualV20 Model = "embed-multilingual-v2.0"
+const EmbedEnglishLightV20 Model = "embed-english-light-v2.0"
+const EmbedEnglishV20 Model = "embed-english-v2.0"
+
+type CohereEmbedder struct {
+	model  Model
+	client *cohere.Client
+}
+
+// New Creates new Cohere Embedder Client
+func New(model Model, apiKey string) *CohereEmbedder {
```

Comment:
Implement `func (h *CohereEmbedder) WithAPIKey(token string) *CohereEmbedder` and  `func (h *CohereEmbedder) WithModel(token string) *CohereEmbedder`.

`New()` will use a default key (the one from os.Getenv("COHERE_API_KEY) and a default model from constants.